client/src/components/lobby/Lobby.js

```diff
@@ -1,34 +1,38 @@
 //@flow
 
-import React, { useEffect, useContext, useState } from "react";
-import type { UserInLobbyT, UserT } from "common/types";
+import React, { useEffect, useContext, useState, useRef } from "react";
+import type { UserInLobbyT } from "common/types";
 import UserList from "./UserList";
 import ReadyButton from "./ReadyButton";
 import PlayButton from "./PlayButton";
 import SocketContext from "contexts/SocketContext";
-
-const mockUsers: Array<UserInLobbyT> = [
-  { id: "id1", username: "username1", roomId: "room", state: "Ready" },
-  { id: "id2", username: "username2", roomId: "room", state: "Waiting" },
-  { id: "id3", username: "username3", roomId: "room", state: "Waiting" },
-  { id: "id4", username: "username4", roomId: "room", state: "Ready" },
-];
+import errorCallBack from "utils/errorCallBack";
 
 const Lobby = (): React$Element<any> => {
   const socket = useContext(SocketContext);
-  const [userList: Array<UserInLobbyT>, setUserList] = useState(mockUsers);
-  const addUserToList = (user: UserT) => {
-    const userInLobby: UserInLobbyT = {
-      ...user,
-      state: "Waiting",
-    };
-    const updatedUserList: Array<UserInLobbyT> = [...userList, userInLobby];
-    setUserList(updatedUserList);
-  };
+  const [userList: Array<UserInLobbyT>, setUserList] = useState([]);
   useEffect(() => {
-    socket.on("user-joined", (user: UserT) => {
-      addUserToList(user);
+    socket.emit("get-users-in-lobby", errorCallBack, (users)=>{
+      setUserList(users);
     });
+  }, []);
```

Comment:
Here you are using socket inside useEffect, this will throw a warning on compilation. Since socket is a constant, you can add it to the dependency array, or you can add `// eslint-disable-line` or `// eslint-disable-next-line` to avoid the warnings.